examples/onebrc/README.md

```diff
@@ -0,0 +1,216 @@
+
+## Proton
+Proton is a purpose-built streaming analytics engine that comes bundled with [two data stores](https://docs.timeplus.com/proton-architecture#data-storage): 
+- Timeplus NativeLog data store for real-time streaming queries and;
+- ClickHouse data store for historical queries.
+
+Since the input data for the `1brc` is a static 13GB CSV file and not a streaming data source, we will simply adapt the solution written in the ClickHouse SQL dialect so it can work inside Proton for this demo. 
+
+### Adapting the ClickHouse SQL for Proton
+The core of the ClickHouse SQL solution[^a] can be seen below:
+```sql
+SET format_csv_delimiter = ';';
+
+SELECT 
+    concat('{', arrayStringConcat(groupArray(formatted_result), ', '), '}') AS final_output
+FROM (
+    SELECT 
+        format('{}={}/{}/{}', city, toDecimalString(min(temperature), 1), toDecimalString(avg(temperature), 1), toDecimalString(max(temperature), 1)) AS formatted_result
+    FROM file('measurements.txt', 'CSV', 'city String, temperature Float32')
+    GROUP BY city
+    ORDER BY city
+)
+```
+
+The Proton docs covers the [minor differences](https://docs.timeplus.com/proton-faq/#if-im-familiar-with-clickhouse-how-easy-is-it-for-me-to-use-proton) between the ClickHouse SQL dialect and the Proton SQL dialect. 
+
+In all, the changes were minimal to make the SQL work on Proton:
+* convert column types from title case to lower case:
+  * in `city String`, change `String` => `string`;
+  * in `temperature Float32`, change `Float32` => `float32`;
+* convert function names from camel case to snake case:
+  * change `arrayStringConcat()` => `array_string_concat()`;
+  * change `groupArray()` => `group_array()`;
+* in the absence of a single function, compose multiple functions to obtain the same result: 
+  * change `toDecimalString()` => `to_string(to_decimal(...))`.
+
+The SQL code that we will be using on Proton:
+```sql
+SET format_csv_delimiter = ';';
+
+SELECT 
+    concat('{', array_string_concat(group_array(formatted_result), ', '), '}') AS final_output
+FROM (
+    SELECT 
+        format('{}={}/{}/{}', city, to_string(to_decimal(min(temperature), 1)), to_string(to_decimal(avg(temperature), 1)), to_string(to_decimal(max(temperature), 1))) AS formatted_result
+    FROM file('measurements.txt', 'CSV', 'city string, temperature float32')
+    GROUP BY city
+    ORDER BY city
+)
+```
+
+## Local Setup
+### Setup the Test Data
+1. Make a separate folder for this demo to make it easy to clean up after we are done:
+```bash
+sudo mkdir -p /demo
+sudo chown -R ubuntu:ubuntu /demo
+```
+
+2. The official test data generator was written for Java 21. We will first install [`sdkman`](https://sdkman.io/jdks) to allow us manage multiple Java versions:
+```bash
+curl -s "https://get.sdkman.io" | bash
+source "/home/ubuntu/.sdkman/bin/sdkman-init.sh"
+```
+
+3. Using `sdkman`, install a JDK based on OpenJDK that supports Java 21:
+```bash
+sdk install java 21.0.1-open
```

Comment:
minor change. `21.0.1-open` should still work, but it's not listed on `sdk ls java` any more.

`sdk install java 21.0.2-open` would be better (but it can also get out-of-date easily)